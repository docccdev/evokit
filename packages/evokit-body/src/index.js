import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
