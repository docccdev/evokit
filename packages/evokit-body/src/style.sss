@import './mixins.sss'

@custom-media --ek-body-media-small only screen and (min-width: 480px)
@custom-media --ek-body-media-medium only screen and (min-width: 768px)
@custom-media --ek-body-media-large only screen and (min-width: 960px)
@custom-media --ek-body-media-wide only screen and (min-width: 1200px)
@custom-media --ek-body-media-huge only screen and (min-width: 1400px)

:root
    // base
    --ek-body-font-family: system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif
    --ek-body-font-size: 13px
    --ek-body-font-weight: 400
    --ek-body-line-height: 1em
    --ek-body-min-width: 320px
    --ek-body-max-width: 1240px
    // prop 'body-indent'
    --ek-body-indent-xxs: 5px
    --ek-body-indent-xs: 10px
    --ek-body-indent-s: 15px
    --ek-body-indent-m: 20px
    --ek-body-indent-l: 25px
    --ek-body-indent-xl: 30px
    --ek-body-indent-xxl: 35px
    --ek-body-indent-3xl: 40px
    --ek-body-indent-4xl: 45px
    --ek-body-indent-5xl: 50px
    // prop 'body-width', 'body-width-min', 'body-width-max'
    --ek-body-width-xxs: 320px
    --ek-body-width-xs: 480px
    --ek-body-width-s: 768px
    --ek-body-width-m: 900px
    --ek-body-width-l: 1024px
    --ek-body-width-xl: 1280px
    --ek-body-width-xxl: 1366px
    --ek-body-width-3xl: 1440px
    --ek-body-width-4xl: 1600px
    --ek-body-width-5xl: 1920px
    // DEPRECATED prop 'body-size'
    --ek-body-max-width-xl: 1440px
    --ek-body-max-width-xxl: 1640px
    --ek-body-max-width-xxxl: 1840px

.body
    display: block
    font-family: var(--ek-body-font-family)
    font-size: var(--ek-body-font-size)
    line-height: var(--ek-body-line-height)
    font-weight: var(--ek-body-font-weight)
    min-width: var(--ek-body-min-width)
    margin: 0
    text-size-adjust: 100%
    -webkit-font-smoothing: subpixel-antialiased

.body__section
    display: block
    box-sizing: border-box
    margin-left: auto
    margin-right: auto
    min-width: var(--ek-body-min-width)
    max-width: var(--ek-body-max-width)

/**
 * Body display:
 *
 *  - body_display_block
 *  - body_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .body_display_$(value)
        display: $value

/**
 * body-size:
 *
 *  - body_size_[xl|xxl|xxxl]
 *
 **/

@each $value in (xl, xxl, xxxl)
    @mixin media .body_size_$(value)
        .body__section
            max-width: var(--ek-body-max-width-$(value))

/**
 * body-width:
 *
 *  - body_width_none
 *  - body_width_xxs
 *  - body_width_xs
 *  - body_width_s
 *  - body_width_m
 *  - body_width_l
 *  - body_width_xl
 *  - body_width_xxl
 *  - body_width_3xl
 *  - body_width_4xl
 *  - body_width_5xl
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl)
    @mixin media .body_width_$(value)
        @if $value == none
            min-width: 0
        @else
            min-width: var(--ek-body-width-$(value))

        .body__section
            @if $value == none
                min-width: 0
                max-width: none
            @else
                min-width: var(--ek-body-width-$(value))
                max-width: var(--ek-body-width-$(value))

    @mixin media .body_width-min_$(value)
        @if $value == none
            min-width: 0
        @else
            min-width: var(--ek-body-width-$(value))

        .body__section
            @if $value == none
                min-width: 0
            @else
                min-width: var(--ek-body-width-$(value))

    @mixin media .body_width-max_$(value)
        .body__section
            @if $value == none
                max-width: none
            @else
                max-width: var(--ek-body-width-$(value))

/**
 * body-indent:
 *
 *  - body_indent_none
 *  - body_indent_xxs
 *  - body_indent_xs
 *  - body_indent_s
 *  - body_indent_m
 *  - body_indent_l
 *  - body_indent_xl
 *  - body_indent_xxl
 *  - body_indent_3xl
 *  - body_indent_4xl
 *  - body_indent_5xl
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl)
    @mixin media .body_indent_$(value)
        .body__section
            @if $value == none
                padding-left: 0
                padding-right: 0
            @else
                padding-left: var(--ek-body-indent-$(value))
                padding-right: var(--ek-body-indent-$(value))

/**
 * Overflow:
 *
 *  - body_overflow_*
 *  - body_overflow-x_*
 *  - body_overflow-y_*
 *
 **/

@each $value in (auto, hidden, scroll, visible)
    @mixin media .body_overflow_$(value)
        overflow: $value

    @mixin media .body_overflow-x_$(value)
        overflow-x: $value

    @mixin media .body_overflow-y_$(value)
        overflow-y: $value

/**
 * body-overflow-anchor:
 *
 *  - body_overflow-anchor_none
 *  - body_overflow-anchor_auto
 *
 **/

@each $value in (none, auto)
    @mixin media .body_overflow-anchor_$(value)
        overflow-anchor: $value

/**
 * Section display:
 *
 *  - body__section_display_block
 *  - body__section_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .body__section_display_$(value)
        display: $value

/**
 * Section height:
 *
 *  - body__section_height_auto
 *  - body__section_height_inherit
 *  - body__section_height_1-1
 *
 **/

@each $value in (auto, inherit, 1-1)
    @mixin media .body__section_height_$(value)
        @if $value == 1-1
            height: 100%
        @else
            height: $value
