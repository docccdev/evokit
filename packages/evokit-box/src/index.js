// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
