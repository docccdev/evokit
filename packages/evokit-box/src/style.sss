@import './mixins.sss'

@custom-media --ek-box-media-small only screen and (min-width: 480px)
@custom-media --ek-box-media-medium only screen and (min-width: 768px)
@custom-media --ek-box-media-large only screen and (min-width: 960px)
@custom-media --ek-box-media-wide only screen and (min-width: 1200px)
@custom-media --ek-box-media-huge only screen and (min-width: 1400px)

:root
    // prop 'box-padding', 'box-margin'
    --ek-box-indent-xxs: 5px
    --ek-box-indent-xs: 10px
    --ek-box-indent-s: 15px
    --ek-box-indent-m: 20px
    --ek-box-indent-l: 25px
    --ek-box-indent-xl: 30px
    --ek-box-indent-xxl: 35px
    // prop 'box-round'
    --ek-box-round-xxs: 2px
    --ek-box-round-xs: 4px
    --ek-box-round-s: 6px
    --ek-box-round-m: 8px
    --ek-box-round-l: 10px
    --ek-box-round-xl: 12px
    --ek-box-round-xxl: 14px
    // prop 'box-border-width'
    --ek-box-border-width-xxs: 1px
    --ek-box-border-width-xs: 2px
    --ek-box-border-width-s: 3px
    --ek-box-border-width-m: 4px
    --ek-box-border-width-l: 5px
    --ek-box-border-width-xl: 6px
    --ek-box-border-width-xxl: 7px

.box
    display: block
    box-sizing: border-box
    border-width: 0
    border-style: solid

    &:after
        clear: both

    &:after, &:before
        content: ''
        display: table

/**
 * Align:
 *
 *  - box_align_none
 *  - box_align_left
 *  - box_align_center
 *  - box_align_right
 *
 **/

@each $value in (none, left, center, right)
    @mixin media .box_align_$(value)
        @if $value == center
            float: none
            display: inline-block
            position: relative
            left: 50%
            transform: translateX(-50%)
        @else
            float: $value
            display: block
            position: static
            left: 0
            transform: translateX(0)

/**
 * Display:
 *
 *  - box_display_*
 *
 **/

@each $value in (none, inline, inline-block, block)
    @mixin media .box_display_$(value)
        display: $value

        @if $value == inline
            &:after, &:before
                content: none

/**
 * Position:
 *
 *  - box_position_*
 *
 **/

@each $value in (static, relative, absolute, fixed)
    @mixin media .box_position_$(value)
        position: $value

/**
 * Place:
 *
 *  - box_place_*
 *
 **/

@each $value in (top, top-left, top-right, center, center-left, center-right, bottom, bottom-left, bottom-right)
    @mixin media .box_place_$(value)
        @if $value == top
            top: 0
            left: 50%
            transform: translateX(-50%)
        @else if $value == top-left
            top: 0
            left: 0
            transform: none
        @else if $value == top-right
            top: 0
            right: 0
            transform: none
        @else if $value == center
            top: 50%
            left: 50%
            transform: translate(-50%, -50%)
        @else if $value == center-left
            top: 50%
            left: 0
            transform: translateY(-50%)
        @else if $value == center-right
            top: 50%
            right: 0
            transform: translateY(-50%)
        @else if $value == bottom
            bottom: 0
            left: 50%
            transform: translateX(-50%)
        @else if $value == bottom-left
            bottom: 0
            left: 0
            transform: none
        @else if $value == bottom-right
            bottom: 0
            right: 0
            transform: none

/**
 * Margin:
 *
 *  - box_margin_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, auto)
    @mixin media .box_margin_$(value)
        @if $value == none
            margin: 0
        @else if $value == auto
            margin: auto
        @else
            margin: var(--ek-box-indent-$(value))

    @mixin media .box_margin-tb_$(value)
        @if $value == none
            margin-top: 0
            margin-bottom: 0
        @else if $value == auto
            margin-top: auto
            margin-bottom: auto
        @else
            margin-top: var(--ek-box-indent-$(value))
            margin-bottom: var(--ek-box-indent-$(value))

    @mixin media .box_margin-lr_$(value)
        @if $value == none
            margin-left: 0
            margin-right: 0
        @else if $value == auto
            margin-left: auto
            margin-right: auto
        @else
            margin-left: var(--ek-box-indent-$(value))
            margin-right: var(--ek-box-indent-$(value))

    @mixin media .box_margin-left_$(value)
        @if $value == none
            margin-left: 0
        @else if $value == auto
            margin-left: auto
        @else
            margin-left: var(--ek-box-indent-$(value))

    @mixin media .box_margin-top_$(value)
        @if $value == none
            margin-top: 0
        @else if $value == auto
            margin-top: auto
        @else
            margin-top: var(--ek-box-indent-$(value))

    @mixin media .box_margin-right_$(value)
        @if $value == none
            margin-right: 0
        @else if $value == auto
            margin-right: auto
        @else
            margin-right: var(--ek-box-indent-$(value))

    @mixin media .box_margin-bottom_$(value)
        @if $value == none
            margin-bottom: 0
        @else if $value == auto
            margin-bottom: auto
        @else
            margin-bottom: var(--ek-box-indent-$(value))

/**
 * Padding:
 *
 *  - box_padding_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl)
    @mixin media .box_padding_$(value)
        @if $value == none
            padding: 0
        @else
            padding: var(--ek-box-indent-$(value))

    @mixin media .box_padding-tb_$(value)
        @if $value == none
            padding-top: 0
            padding-bottom: 0
        @else
            padding-top: var(--ek-box-indent-$(value))
            padding-bottom: var(--ek-box-indent-$(value))

    @mixin media .box_padding-lr_$(value)
        @if $value == none
            padding-left: 0
            padding-right: 0
        @else
            padding-left: var(--ek-box-indent-$(value))
            padding-right: var(--ek-box-indent-$(value))

    @mixin media .box_padding-left_$(value)
        @if $value == none
            padding-left: 0
        @else
            padding-left: var(--ek-box-indent-$(value))

    @mixin media .box_padding-top_$(value)
        @if $value == none
            padding-top: 0
        @else
            padding-top: var(--ek-box-indent-$(value))

    @mixin media .box_padding-right_$(value)
        @if $value == none
            padding-right: 0
        @else
            padding-right: var(--ek-box-indent-$(value))

    @mixin media .box_padding-bottom_$(value)
        @if $value == none
            padding-bottom: 0
        @else
            padding-bottom: var(--ek-box-indent-$(value))

/**
 * Round:
 *
 *  - box_round_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, full)
    @mixin media .box_round_$(value)
        @if $value == none
            border-radius: 0
        @else if $value == full
            border-radius: 50%
        @else
            border-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-top_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-top-right-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-top-right-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))
            border-top-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-right_$(value)
        @if $value == none
            border-top-right-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-box-round-$(value))
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-bottom_$(value)
        @if $value == none
            border-bottom-left-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-box-round-$(value))
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-left_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-bottom-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-bottom-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))
            border-bottom-left-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-top-left_$(value)
        @if $value == none
            border-top-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-top-right_$(value)
        @if $value == none
            border-top-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-bottom-right_$(value)
        @if $value == none
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-right-radius: 50%
        @else
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-bottom-left_$(value)
        @if $value == none
            border-bottom-left-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-box-round-$(value))


/**
 * Width:
 *
 *  - box_width_*
 *
 **/

@each $to in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
    @each $from in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
        @if $from < $to or ($from == 1 and $to == 1)
            @mixin media .box_width_$(from)-$(to)
                width: calc($from / $to * 100%)

@mixin media .box_width_auto
    width: auto

/**
 * Height:
 *
 *  - box_height_*
 *
 **/

@mixin media .box_height_1-1
    height: 100%

@mixin media .box_height_auto
    height: auto

/**
 * Overflow:
 *
 *  - box_overflow_*
 *  - box_overflow-x_*
 *  - box_overflow-y_*
 *
 **/

@each $value in (auto, hidden, scroll, visible)
    @mixin media .box_overflow_$(value)
        overflow: $value

    @mixin media .box_overflow-x_$(value)
        overflow-x: $value

    @mixin media .box_overflow-y_$(value)
        overflow-y: $value
