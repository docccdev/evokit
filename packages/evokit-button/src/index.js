import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
