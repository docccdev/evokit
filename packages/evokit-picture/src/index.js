import { createBlock } from 'evokit';

export const Picture = createBlock('picture', 'picture', [
    'background',
    'display',
    'fit',
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'border-lr',
    'border-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);

export const PictureItem = createBlock('img', 'picture__item', ['display']);

Picture.Item = PictureItem;
