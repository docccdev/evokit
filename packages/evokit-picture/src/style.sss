@import './mixins.sss'

@custom-media --ek-picture-media-small only screen and (min-width: 480px)
@custom-media --ek-picture-media-medium only screen and (min-width: 768px)
@custom-media --ek-picture-media-large only screen and (min-width: 960px)
@custom-media --ek-picture-media-wide only screen and (min-width: 1200px)
@custom-media --ek-picture-media-huge only screen and (min-width: 1400px)

:root
    // prop 'picture-round'
    --ek-picture-round-xxs: 2px
    --ek-picture-round-xs: 4px
    --ek-picture-round-s: 6px
    --ek-picture-round-m: 8px
    --ek-picture-round-l: 10px
    --ek-picture-round-xl: 12px
    --ek-picture-round-xxl: 14px

.picture
    display: block
    position: relative
    padding-top: 100%
    overflow: hidden

.picture__item
    position: absolute
    top: 0
    right: 0
    bottom: 0
    left: 0
    max-width: 100%
    max-height: 100%
    margin: auto
    outline: none
    border: 0

    &[alt]
        word-wrap: break-word
        word-break: break-word
        hyphens: auto

/**
 * Display:
 *
 *  - picture_display_block
 *  - picture_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .picture_display_$(value)
        display: $value

/**
 * Item display:
 *
 *  - picture__item_display_block
 *  - picture__item_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .picture__item_display_$(value)
        display: $value

/**
 * Fit:
 *
 *  - picture_fit_*
 *
 **/

@each $value in (none, fill, contain, cover, scale-down)
    @mixin media .picture_fit_$(value)
        transform: translatez(0)

        & > .picture__item
            object-fit: $value
            width: 100%
            height: 100%

/**
 * Round:
 *
 *  - picture_round_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, full)
    @mixin media .picture_round_$(value)
        @if $value == none
            border-radius: 0
        @else if $value == full
            border-radius: 50%
        @else
            border-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-top_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-top-right-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-top-right-radius: 50%
        @else
            border-top-left-radius: var(--ek-picture-round-$(value))
            border-top-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-right_$(value)
        @if $value == none
            border-top-right-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-picture-round-$(value))
            border-bottom-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-bottom_$(value)
        @if $value == none
            border-bottom-left-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-picture-round-$(value))
            border-bottom-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-left_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-bottom-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-bottom-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-picture-round-$(value))
            border-bottom-left-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-top-left_$(value)
        @if $value == none
            border-top-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-top-right_$(value)
        @if $value == none
            border-top-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-bottom-right_$(value)
        @if $value == none
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-right-radius: 50%
        @else
            border-bottom-right-radius: var(--ek-picture-round-$(value))

    @mixin media .picture_round-bottom-left_$(value)
        @if $value == none
            border-bottom-left-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-picture-round-$(value))
