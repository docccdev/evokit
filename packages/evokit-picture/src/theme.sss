@import './mixins.sss'

:root
    --ek-picture-border-TEMPLATE_NAME: var(--border)

@mixin media .picture_border_TEMPLATE_NAME
    border: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-top_TEMPLATE_NAME
    border-top: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-right_TEMPLATE_NAME
    border-right: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-bottom_TEMPLATE_NAME
    border-bottom: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-left_TEMPLATE_NAME
    border-left: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-tb_TEMPLATE_NAME
    border-top: 1px solid var(--ek-picture-border-TEMPLATE_NAME)
    border-bottom: 1px solid var(--ek-picture-border-TEMPLATE_NAME)

@mixin media .picture_border-lr_TEMPLATE_NAME
    border-left: 1px solid var(--ek-picture-border-TEMPLATE_NAME)
    border-right: 1px solid var(--ek-picture-border-TEMPLATE_NAME)
