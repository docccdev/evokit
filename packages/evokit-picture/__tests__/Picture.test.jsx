import React from 'react';
import { shallow } from 'enzyme';
import barista from 'seed-barista';
import style from '../style.css';
import { Picture, PictureItem } from '../src';

describe('<Picture />', () => {
    it('render', () => {
        const wrapper = shallow(<Picture />);
        const wrapperItem = shallow(<PictureItem />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapperItem).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<picture class="ek-picture"></picture>');
        expect(wrapperItem.html()).toEqual('<img class="ek-picture__item"/>');
    });

    it('props className', () => {
        const wrapper = shallow(<Picture className='ololo' />);
        const wrapperItem = shallow(<PictureItem className='ololo' />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapperItem).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<picture class="ek-picture ololo"></picture>');
        expect(wrapperItem.html()).toEqual('<img class="ek-picture__item ololo"/>');
    });

    it('props picture-tag', () => {
        const wrapper = shallow(<Picture picture-tag='span' />);
        const wrapperItem = shallow(<PictureItem picture-item-tag='span' />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapperItem).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<span class="ek-picture"></span>');
        expect(wrapperItem.html()).toEqual('<span class="ek-picture__item"></span>');
    });

    it('css', () => {
        const css = barista({ content: style });

        expect(css.rule('.ek-picture').exists()).toBeTruthy();
        expect(css.rule('.ek-picture:after').exists()).toBeTruthy();
        expect(css.rule('.ek-picture__item').exists()).toBeTruthy();
        expect(css.rule('.ek-picture__item[alt]').exists()).toBeTruthy();
    });
});
