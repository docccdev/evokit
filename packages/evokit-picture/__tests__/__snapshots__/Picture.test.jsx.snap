// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Picture /> props className 1`] = `
<picture
  className="ek-picture ololo"
/>
`;

exports[`<Picture /> props className 2`] = `
<img
  className="ek-picture__item ololo"
/>
`;

exports[`<Picture /> props picture-tag 1`] = `
<span
  className="ek-picture"
/>
`;

exports[`<Picture /> props picture-tag 2`] = `
<span
  className="ek-picture__item"
/>
`;

exports[`<Picture /> render 1`] = `
<picture
  className="ek-picture"
/>
`;

exports[`<Picture /> render 2`] = `
<img
  className="ek-picture__item"
/>
`;
